cluster/addons/kube-proxy/kube-proxy-ds.yaml

```diff
@@ -48,7 +48,9 @@ spec:
         {{kube_cache_mutation_detector_env_name}}
           {{kube_cache_mutation_detector_env_value}}
         securityContext:
-          privileged: true
+          capabiltiies:
+            add:
+            - NET_ADMIN 
```

Comment:
something tells me that this isn't sufficient, have you tested the change in all kube-proxy modes?
AFAIK, `NET_RAW` should be required for "iptables" too.

@kubernetes/sig-network-pr-reviews

Reply:
the `iptables -w` option requires mounting `/run/xtables.lock` which needs root.
https://github.com/kubernetes/kubernetes/issues/46103
https://github.com/kubernetes/kubernetes/pull/46372

Reply:
IIRC, the UID in the container is 0.  Please remind me, mounting a host path shouldn't need CAP_SYS_ADMIN.

To answer your earlier question, no, I have not tested this in all kube-proxy modes.  Was having a little trouble getting me dev env setup.  I'll try harder and test more locally.

Reply:
> IIRC, the UID in the container is 0. Please remind me, mounting a host path shouldn't need CAP_SYS_ADMIN.

it shouldn't. ok, i guess the mounting is not a problem here.

> To answer your earlier question, no, I have not tested this in all kube-proxy modes. Was having a little trouble getting me dev env setup. I'll try harder and test more locally.

if this PR gets stuck without review, you could ask for a discussion in #sig-network on k8s slack.

Reply:
cool, thanks!